src/core/server/http/lifecycle/auth.ts

```diff
@@ -16,33 +16,25 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import Boom from 'boom';
-import { noop } from 'lodash';
 import { Lifecycle, Request, ResponseToolkit } from 'hapi';
-import { KibanaRequest } from '../router';
+import { Logger } from '../../logging';
+import {
+  HapiResponseAdapter,
+  KibanaRequest,
+  KibanaResponse,
+  lifecycleResponseFactory,
+  LifecycleResponseFactory,
+} from '../router';
 
 enum ResultType {
   authenticated = 'authenticated',
-  redirected = 'redirected',
-  rejected = 'rejected',
 }
 
 interface Authenticated extends AuthResultParams {
   type: ResultType.authenticated;
 }
 
-interface Redirected {
-  type: ResultType.redirected;
-  url: string;
-}
-
-interface Rejected {
-  type: ResultType.rejected;
-  error: Error;
-  statusCode?: number;
-}
-
-type AuthResult = Authenticated | Rejected | Redirected;
+type AuthResult = Authenticated;
```

Comment:
It can be as simple as Authenticated. The current implementation hides details and allows us to introduce additional logic without affecting the interface.